pkg/s3select/select.go

```
@@ -153,143 +143,170 @@ func ParseSelect(f format.Select) ([]string, string, int64, interface{}, []strin
 		if err := parseErrs(columnNames, whereClause, alias, sFuncs, f); err != nil {
 			return nil, "", 0, nil, nil, sFuncs, err
 		}
-		return columnNames, alias, limit, whereClause, functionNames, sFuncs, nil
+		return columnNames, alias, limit, whereClause, fnNames, sFuncs, nil
 	}
 	return nil, "", 0, nil, nil, sFuncs, nil
 }
 
+type columnKv struct {
+	Key   string
+	Value int
+}
+
+func columnsIndex(reqColNames []string, f format.Select) ([]columnKv, error) {
+	var (
+		columnsKv  []columnKv
+		columnsMap = make(map[string]int)
+		columns    = f.Header()
+	)
+	if f.HasHeader() {
```

Can we handle the case when reqColNames is empty before starting the real code ?

---

Not sure what you mean.

---

> Not sure what you mean.

We need to return an error when reqColNames is empty, otherwise it will crash

---

@vadmeste reqColNames is never empty

---

> @vadmeste reqColNames is never empty

Currently it can happen if you run `mc select --query "select inexistant_colum from S3Object;" myminio/testbucket/file.csv`, we probably the fix needs to be somewhere else

---

This is fixed..